src/uct/ib/rc/base/rc_iface.h

```diff
@@ -513,6 +513,18 @@ uct_rc_iface_check_pending(uct_rc_iface_t *iface, ucs_arbiter_group_t *arb_group
     ucs_assert(iface->tx.in_pending || ucs_arbiter_group_is_empty(arb_group));
 }
 
+static UCS_F_ALWAYS_INLINE void
+uct_rc_iface_check_pending_ep_flush_cancel(uct_rc_iface_t *iface,
+                                           ucs_arbiter_group_t *arb_group)
+{
+    /* Allow flush(CANCEL) when EP's arbiter group is empty or flushing from
+     * a pending callback and only element responsible for flushing is scheduled
+     * on an arbiter */
+    ucs_assert(ucs_arbiter_group_is_empty(arb_group) ||
+               (iface->tx.in_pending &&
+                (ucs_arbiter_group_num_elems(arb_group) == 1)));
```

Comment:
i don't think we can make such assumption, we can't really know if the pending element(s) is going to be flush (cancel)
better remove this assert altogether, and replace by assertion that we don't post_send anything, and don't flush(LOCAL) after the QP is in error state (could be in other PR)

Reply:
I guess flush(CANCEL) requires to purge all pending element prior calling it

Reply:
not sure it works if we call it from pending queue and already have few pending operations
maybe - instead - we can check that CANCEL flag on the ep is not set, during *pending_add*?

Reply:
we call it from the pending queue, since we do it in discarding (discarding purges all prnding operation prior)
we tried flush(CANCEL) - but if returns `UCS_ERR_NO_RESOURCE` and we put it to pending queue and then try again from pending callback
> maybe - instead - we can check that CANCEL flag on the ep is not set, during pending_add?
it is a good idea, we could do it as well